File: sol/3289/3289.cpp
struct Solution {
    vector<int> getSneakyNumbers(vector<int>& nums) {
        vector<int> res;
        for (int x : nums){
            if (x < 0){
                x = -x - 1;
            }
            if (nums[x] < 0){
                res.push_back(x);
            } else {
                nums[x] = -nums[x] - 1;
            }
        }
        return res;
    }
};

File: sol/3289/3289.py
class Solution:
    def getSneakyNumbers(self, nums: List[int]) -> List[int]:
        res = []
        for x in nums:
            if x < 0:
                x = -x - 1
            if nums[x] < 0:
                res.append(x)
            else:
                nums[x] = -nums[x] - 1
        return res
